index.bs

```diff
@@ -0,0 +1,1016 @@
+        element, then the more restrictive allowlist of
+        <code>allow="payment"</code> will be used.
+      </div>
+    </section>
+  </section>
+</section>
+<section>
+  <h2 id="integrations">Integrations</h2>
+  <p>This document defines a set of algorithms which other specifications will
+  use in order to implement the restrictions which Feature Policy defines. The
+  integrations are outlined here for clarity, but those external documents are
+  the normative references which ought to be consulted for detailed
+  information.</p>
+  <section>
+    <h3 id="integration-with-html">Integration with HTML</h3>
+    <ol>
+      <li>
+        {{Document}} and {{WorkerGlobalScope}} objects have a
+        <a>Feature Policy</a>, which is populated via the <a href=
+        "#initialize-for-global"></a> algorithm that is called during the
+        "Initialising a new <code>Document</code> object" and "Run a Worker"
+        algorithms.
+      </li>
+      <li>Replace the existing step 12 of "Initialising a new
+      <code>Document</code> object" with the following step:
+        <ul>
+          <li>
+            <a href="#initialize-for-global">Initialize the feature policy</a>
+            for the <code>Document</code>
+          </li>
+        </ul>
+      </li>
+      <li>A <a>feature policy</a> is <dfn data-lt="enforce">enforced</dfn> for
+      a {{Document}} or {{WorkerGlobalScope}} by setting it as the {{Document}}
+      or {{WorkerGlobalScope}}'s <a>Feature Policy</a>.
+      </li>
+      <li>
+        <p>The "[=allowed to use=]" algorithm calls into <a href=
+        "#is-feature-enabled"></a>, as follows:</p>
+        <ol>
+          <li>Replace the current steps #3 and #4 with the following step:
+            <ul>
+              <li>If <code>Document</code>'s <a>feature policy</a> enables the
+              feature indicated by <code>allowattribute</code> for the origin
+              of <code>Document</code>, then return true.
+              </li>
+            </ul>
+          </li>
+        </ol>
+      </li>
+    </ol>
+    <div class="issue">
+      Monkey-patching! As soon as we know that this is the direction we wish to
+      pursue, upstream all of this.
+    </div>
+    <section>
+      <h4 id="process-meta-policy">Process meta policy</h4>
+      <div class="note">
+        This section extends HTML's <code>meta</code> [pragma directives].
+      </div><strong>Feature policy state (http-equiv="feature-policy")</strong>
+      <p>This pragma extends the declared feature policy for a {{Document}}.</p>
+      <ol>
+        <li>If the [meta] element has no [content] attribute, or if that
+        attribute's value is the empty string, then abort these steps.</li>
+        <li>If the [meta] element is not being inserted in into the
+        {{Document}}'s [head] element, or if is inserted after any [script] or
+        [link] elements, then abort these steps.</li>
```

Comment:
That's an interesting possibility -- it would only allow a single policy, and would mean that we abandon the http-equiv idea, but it would certainly be parseable before any scripts get a chance to run.

I'll remove the <meta> bits from the spec, and bring this up in an issue separately then. Having the ability to set the policy for the top-level frame purely in HTML is nice-to-have, but not, I think, strictly necessary, with the set of features currently listed.